kratos/tests/cpp_tests/utilities/test_divide_triangle_3d_3.cpp

```diff
@@ -0,0 +1,202 @@
+		{
+			Model current_model;
+
+			// Generate a model part with the previous
+			ModelPart& base_model_part = current_model.CreateModelPart("Triangle");
+			base_model_part.AddNodalSolutionStepVariable(DISTANCE);
+
+			// Fill the model part geometry data
+			base_model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+			base_model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+			base_model_part.CreateNewNode(3, 0.0, 1.0, -1.0);
+			Properties::Pointer p_properties(new Properties(0));
+			base_model_part.CreateNewElement("Element3D3N", 1, {1, 2, 3}, p_properties);
+
+			// Set the DISTANCE field
+			base_model_part.Nodes()[1].FastGetSolutionStepValue(DISTANCE) = -1.0;
+			base_model_part.Nodes()[2].FastGetSolutionStepValue(DISTANCE) = -1.0;
+			base_model_part.Nodes()[3].FastGetSolutionStepValue(DISTANCE) =  1.0;
+
+			// Set the elemental distances vector
+			Geometry < Node < 3 > >& r_geometry = base_model_part.Elements()[1].GetGeometry();
+
+			array_1d<double, 3> distances_vector;
+			for (unsigned int i = 0; i < r_geometry.size(); ++i) {
+				distances_vector(i) = r_geometry[i].FastGetSolutionStepValue(DISTANCE);
+			}
+
+			base_model_part.Elements()[1].SetValue(ELEMENTAL_DISTANCES, distances_vector);
+
+			Vector& r_elemental_distances = base_model_part.Elements()[1].GetValue(ELEMENTAL_DISTANCES);
+
+			// Build the triangle splitting utility
+			DivideTriangle3D3 triangle_splitter(r_geometry, r_elemental_distances);
+
+			// Call the divide geometry method
+			triangle_splitter.GenerateDivision();
+
+
+			// Call the intersection generation method
+			triangle_splitter.GenerateIntersectionsSkin();
+
+			// Call the positive exterior faces generation method
+			std::vector < unsigned int > pos_ext_faces_parent_ids;
+			std::vector < DivideTriangle2D3::IndexedPointGeometryPointerType > pos_ext_faces;
+			triangle_splitter.GenerateExteriorFaces(
+				pos_ext_faces,
+				pos_ext_faces_parent_ids,
+				triangle_splitter.GetPositiveSubdivisions());
+
+			// Call the negative exterior faces generation method
+			std::vector < unsigned int > neg_ext_faces_parent_ids;
+			std::vector < DivideTriangle2D3::IndexedPointGeometryPointerType > neg_ext_faces;
+			triangle_splitter.GenerateExteriorFaces(
+				neg_ext_faces,
+				neg_ext_faces_parent_ids,
+				triangle_splitter.GetNegativeSubdivisions());
+
+			const double tolerance = 1e-10;
+
+			// Check general splitting values
+			KRATOS_CHECK(triangle_splitter.mIsSplit);
+			KRATOS_CHECK_EQUAL(triangle_splitter.mDivisionsNumber, 3);
+			KRATOS_CHECK_EQUAL(triangle_splitter.mSplitEdgesNumber, 2);
+
+			// Check split edges
+			KRATOS_CHECK_EQUAL(triangle_splitter.mSplitEdges[0],  0);
+			KRATOS_CHECK_EQUAL(triangle_splitter.mSplitEdges[1],  1);
+			KRATOS_CHECK_EQUAL(triangle_splitter.mSplitEdges[2],  2);
+			KRATOS_CHECK_EQUAL(triangle_splitter.mSplitEdges[3], -1);
+			KRATOS_CHECK_EQUAL(triangle_splitter.mSplitEdges[4],  4);
+			KRATOS_CHECK_EQUAL(triangle_splitter.mSplitEdges[5],  5);
```

Comment:
Please use `KRATOS_CHECK_VECTOR_EQUAL` as macros increase compilation time a lot (in the days we created the tests for the 2D triangle these were not available).

Reply:
KRATOS_CHECK_VECTOR_EQUAL uses KRATOS_CHECK_VECTOR_NEAR , so it compares against a double.. I think it'll be better to keep it like this.

Reply:
Then you can save the expected values in an `std::array<int,6>` array to then call the `KRATOS_CHECK_EQUAL` within a loop.